sdk/docs/manually-written/sdk/component-howtos/application-development/daml-codegen-java.rst

```diff
@@ -1,37 +1,134 @@
-.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-.. SPDX-License-Identifier: Apache-2.0
-
 .. _daml-codegen-java:
 
+Daml Codegen for Java
+=====================
+
+This tool is part of the Daml Assistant and is used to generate Java classes for Daml data types.
+
+Overview
+-------------
+You can use the Daml codegen to generate Java classes representing Daml contract templates.
+These classes incorporate all boilerplate code for constructing corresponding ledger ``com.daml.ledger.api.v1.CreateCommand``,
+``com.daml.ledger.api.v1.ExerciseCommand``, ``com.daml.ledger.api.v1.ExerciseByKeyCommand``, and ``com.daml.ledger.api.v1.CreateAndExerciseCommand``.
+
+Download
+-------------
+
+The Daml codegen is part of the Daml Assistant. :ref:`Download the Daml Assistant <assistant-manual-managing-releases>`.
+
+.. TODO(#347) Fix the link to link to Download the Daml Assistant of "Daml Assistant"
+
+Install
+-------------
+
+The Daml codegen is part of the Daml Assistant.
```

Comment:
```suggestion
Install the Daml Codegen for Java by `installing the Daml Assistant <link-to-daml-assitant-install>`_.
```

(I believe the above is the right syntax for links. The one for the Daml assistant should already be available.)